lib/Core/Inference/SymbolContextResolver.php

```diff
@@ -597,7 +601,23 @@ private function resolveSubscriptExpression(
 
     private function resolveScopedPropertyAccessExpression(Frame $frame, ScopedPropertyAccessExpression $node): SymbolContext
     {
-        $name = $node->scopeResolutionQualifier->getText();
+        $name = null;
+        if ($node->scopeResolutionQualifier instanceof ParserVariable) {
+            /** @var SymbolContext $context */
+            $context = $this->resolveVariable($frame, $node->scopeResolutionQualifier);
+            if (
+                $context->types() !== null &&
+                $context->types()->count() > 0 &&
+                $context->type()->className() !== null
```

Comment:
`SymbolContext->types()` cannot be `null`, neither can `type()` (it returns a Null object). It should suffice to just just if `className` is `!=null`